src/views/checkups/CheckupsRunButton.tsx

```diff
@@ -9,50 +9,64 @@ import { Button, ButtonVariant } from '@patternfly/react-core';
 
 import useCheckupsNetworkPermissions from './network/hooks/useCheckupsNetworkPermissions';
+import useCheckupsSelfValidationPermissions from './self-validation/components/hooks/useCheckupsSelfValidationPermissions';
 import { useCheckupsStoragePermissions } from './storage/components/hooks/useCheckupsStoragePermissions';
+import { CHECKUP_URLS } from './utils/constants';
+import { getCheckUpTabs } from './utils/getCheckUpsTabs';
 import { trimLastHistoryPath } from './utils/utils';
 
 const CheckupsRunButton: FC = () => {
   const [namespace] = useActiveNamespace();
   const navigate = useNavigate();
   const location = useLocation();
   const { t } = useKubevirtTranslation();
+
   const { isPermitted: isCreateNetworkPermitted } = useCheckupsNetworkPermissions();
   const { isPermitted: isCreateStoragePermitted } = useCheckupsStoragePermissions();
+  const { isPermitted: isCreateSelfValidationPermitted } = useCheckupsSelfValidationPermissions();
+
+  const tabs = useMemo(() => getCheckUpTabs(t), [t]);
+
+  const getPermissionForTab = useCallback(
+    (url: string) => {
+      switch (url) {
+        case CHECKUP_URLS.NETWORK:
+          return isCreateNetworkPermitted;
+        case CHECKUP_URLS.STORAGE:
+          return isCreateStoragePermitted;
+        case CHECKUP_URLS.SELF_VALIDATION:
+          return isCreateSelfValidationPermitted;
+        default:
+          return false;
+      }
+    },
+    [isCreateNetworkPermitted, isCreateStoragePermitted, isCreateSelfValidationPermitted],
+  );
+
+  const createItems = useMemo(() => {
+    const items: Record<string, React.ReactNode> = {};
+
+    tabs.forEach((tab) => {
+      const isPermitted = getPermissionForTab(tab.url);
+
+      items[tab.url] = (
+        <div className={classNames({ 'CheckupsRunButton--item__disabled': !isPermitted })}>
+          {tab.title}
+        </div>
+      );
+    });
 
-  const createItems = {
-    network: (
-      <div
-        className={classNames({ 'CheckupsRunButton--item__disabled': !isCreateNetworkPermitted })}
-      >
-        {t('Network latency')}
-      </div>
-    ),
-    storage: (
-      <div
-        className={classNames({ 'CheckupsRunButton--item__disabled': !isCreateStoragePermitted })}
-      >
-        {t('Storage')}
-      </div>
-    ),
-  };
+    return items;
+  }, [tabs, getPermissionForTab]);
 
   const onCreate = useCallback(
     (type: string) => {
-      switch (type) {
-        case 'network':
-          return (
-            isCreateNetworkPermitted &&
-            navigate(createURL('network/form', trimLastHistoryPath(location.pathname)))
-          );
-        case 'storage':
-          return (
-            isCreateStoragePermitted &&
-            navigate(createURL('storage/form', trimLastHistoryPath(location.pathname)))
-          );
+      const isPermitted = getPermissionForTab(type);
+      if (isPermitted) {
+        navigate(createURL(`${type}/form`, trimLastHistoryPath(location.pathname)));
```

Comment:
it's not a regression but trimming the path and building the url is buggy i.e. try creating a storage checkup from  `/k8s/ns/<ns_name>/checkups/storage/` (note the ending slash)